PWGHF/TableProducer/HFTreeCreatorLcToPKPi.cxx

```diff
@@ -307,6 +307,116 @@ struct CandidateTreeWriter {
+    for (auto& collision : collisions) {
+      rowCandidateFullEvents(
+        collision.bcId(),
+        collision.numContrib(),
+        collision.posX(),
+        collision.posY(),
+        collision.posZ(),
+        0,
+        1);
+    }
+
+    // Filling candidate properties
+    rowCandidateFull.reserve(candidates.size());
+    for (auto& candidate : candidates) {
+      auto fillTable = [&](int CandFlag,
+                           int FunctionSelection,
+                           float FunctionInvMass,
+                           float FunctionCt,
+                           float FunctionY,
+                           float FunctionE) {
+        if (FunctionSelection >= 1) {
+          rowCandidateFull(
+            candidate.index0_as<aod::BigTracksPID>().collision().bcId(),
+            candidate.index0_as<aod::BigTracksPID>().collision().numContrib(),
+            candidate.posX(),
+            candidate.posY(),
+            candidate.posZ(),
+            candidate.xSecondaryVertex(),
+            candidate.ySecondaryVertex(),
+            candidate.zSecondaryVertex(),
+            candidate.errorDecayLength(),
+            candidate.errorDecayLengthXY(),
+            candidate.chi2PCA(),
+            candidate.rSecondaryVertex(),
+            candidate.decayLength(),
+            candidate.decayLengthXY(),
+            candidate.decayLengthNormalised(),
+            candidate.decayLengthXYNormalised(),
+            candidate.impactParameterNormalised0(),
+            candidate.ptProng0(),
+            RecoDecay::P(candidate.pxProng0(), candidate.pyProng0(), candidate.pzProng0()),
+            candidate.impactParameterNormalised1(),
+            candidate.ptProng1(),
+            RecoDecay::P(candidate.pxProng1(), candidate.pyProng1(), candidate.pzProng1()),
+            candidate.impactParameterNormalised2(),
+            candidate.ptProng2(),
+            RecoDecay::P(candidate.pxProng2(), candidate.pyProng2(), candidate.pzProng2()),
+            candidate.pxProng0(),
+            candidate.pyProng0(),
+            candidate.pzProng0(),
+            candidate.pxProng1(),
+            candidate.pyProng1(),
+            candidate.pzProng1(),
+            candidate.pxProng2(),
+            candidate.pyProng2(),
+            candidate.pzProng2(),
+            candidate.impactParameter0(),
+            candidate.impactParameter1(),
+            candidate.impactParameter2(),
+            candidate.errorImpactParameter0(),
+            candidate.errorImpactParameter1(),
+            candidate.errorImpactParameter2(),
+            candidate.index0_as<aod::BigTracksPID>().tpcNSigmaPi(),
+            candidate.index0_as<aod::BigTracksPID>().tpcNSigmaKa(),
+            candidate.index0_as<aod::BigTracksPID>().tpcNSigmaPr(),
+            candidate.index0_as<aod::BigTracksPID>().tofNSigmaPi(),
+            candidate.index0_as<aod::BigTracksPID>().tofNSigmaKa(),
+            candidate.index0_as<aod::BigTracksPID>().tofNSigmaPr(),
```

Comment:
Avoid multiple calls of `index0_as`. It is expensive.

Reply:
Dear Vit,
thanks for pointing this out. I replaced this by calling it once before the table is filled with the properties.